src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -173,6 +180,50 @@ SimpleNameSyntax syntax
 			}
 		}
 
+		private static void AnalyzeConditionalImmutability( SyntaxNodeAnalysisContext ctx ) {
+			var syntax = ctx.Node;
+
+			// Get the symbol for the method
+			if( ctx.SemanticModel.GetDeclaredSymbol( ctx.Node ) is not IMethodSymbol symbol ) {
+				return;
+			}
+
+			// Exit if this somehow is not possible
+			// We don't care about arguments so throw them out
+			if( !GetTypeParamsAndArgs( symbol, out var typeParameters, out _ ) ) {
+				return;
+			}

```

Comment:
```suggestion
```

No need for this call/check, we know we've got an `IMethodSymbol` which has a `TypeParameters`, and as noted we don't need the arguments.

Reply:
Ah I read this wrong, give me a second to update it again...